tests/ui/slice_iter_any.rs

```diff
@@ -0,0 +1,36 @@
+#![warn(clippy::slice_iter_any)]
+
+fn main() {
+    let vec: Vec<u8> = vec![1, 2, 3, 4, 5, 6];
+    let values = &vec[..];
+    let _ = values.iter().any(|&v| v == 4);
+    //~^ ERROR: using `contains()` instead of `iter().any()` on u8/i8 slices is more efficient
+
+    let vec: Vec<u8> = vec![1, 2, 3, 4, 5, 6];
+    let values = &vec[..];
+    let _ = values.contains(&4);
+    // no error, because it uses `contains()`
+
+    let vec: Vec<u32> = vec![1, 2, 3, 4, 5, 6];
+    let values = &vec[..];
+    let _ = values.iter().any(|&v| v == 4);
+    // no error, because it's not a slice of u8/i8
```

Comment:
Wouldn't it be more readable to use `.contains()` here too? I thought that was what @Manishearth was suggesting.